proposals/csharp-11.0/low-level-struct-improvements.md

```diff
@@ -411,6 +410,30 @@ Span<int> ComplexScopedRefExample(scoped ref Span<int> span)
 }
 ```
 
+#### Rules for object initializers
+
+The *safe-context* of an object initializer expression is narrowest of:
+
+1. The *safe-context* of the constructor call.
+2. The *safe-context* and *ref-safe-context* of arguments to member initializer indexers that can escape to the receiver.
+3. The *safe-context* of the RHS of assignments in member initializers to non-readonly setters or *ref-safe-context* in case of ref assignment.
+
+Another way of modeling this is to think of any argument to a member initializer that can be assigned to the receiver as being an argument to the constructor. This is because the member initializer is effectively a constructor call.
+
+```c#
+Span<int> heapSpan = default;
+Span<int> stackSpan = stackalloc int[42];
+var x = new S(ref heapSpan)
+{
+    Field = stackSpan;
+}
+
+// Can be modeled as 
+var x = new S(ref heapSpan, stackSpan)
```

Comment:
Cannot it be modeled simply as this?

```cs
var x = new S(ref heapSpan);
x.Field = stackSpan;
```

Reply:
Doing it that way would mean that the *safe-context* of `x` is *calling method* because only `heapSpan` is used in the variable declaration. The motivation for putting all of the other values in the constructor is to solve issues like this.